packages/vscode-extension/src/debug/prerequisitesHandler.ts

```diff
@@ -263,55 +291,62 @@ async function checkPort(
-    const node: DepsType = await detectNodeDepsType();
-    // node
-    const checkResults: CheckResult[] = [];
-    const nodeResult = await checkNode(
-      node,
-      [node],
-      depsManager,
-      `(${currentStep++}/${totalSteps})`,
-      undefined
-    );
-    if (nodeResult) {
-      checkResults.push(nodeResult);
-    }
-
-    await handleCheckResults(checkResults, undefined, false);
-  } catch (error: any) {
-    const fxError = assembleError(error);
-    showError(fxError);
-    ExtTelemetry.sendTelemetryErrorEvent(TelemetryEvent.GetStartedPrerequisites, fxError);
-    return err(fxError);
+export async function checkPrerequisitesForGetStarted(): Promise<Result<void, FxError>> {
+  const node: DepsType = await detectNodeDepsType();
+  ExtTelemetry.sendTelemetryEvent(TelemetryEvent.GetStartedPrerequisitesStart);
+  const res = await _checkAndInstall(
+    "Prerequisite Check",
+    [{ checker: node, fastFail: false }],
+    []
+  );
+  if (res.error) {
+    ExtTelemetry.sendTelemetryErrorEvent(TelemetryEvent.GetStartedPrerequisites, res.error);
+    return err(res.error);
   }
-  return ok(null);
+  return ok(undefined);
 }
 
 export async function checkAndInstall(): Promise<Result<void, FxError>> {
   const projectComponents = await commonUtils.getProjectComponents();
+  const workspacePath = globalVariables.workspaceUri!.fsPath;
+  const localEnvManager = new LocalEnvManager(VsCodeLogInstance, ExtTelemetry.reporter, VS_CODE_UI);
+  const projectSettings = await localEnvManager.getProjectSettings(workspacePath);
+  const ports = await localEnvManager.getPortsFromProject(workspacePath, projectSettings);
+  const orderedCheckers = await getOrderedCheckers(projectSettings, localEnvManager);
+
   return await localTelemetryReporter.runWithTelemetryProperties(
     TelemetryEvent.DebugPrerequisites,
     // projectComponents is already serialized JSON string
     { [TelemetryProperty.DebugProjectComponents]: `${projectComponents}` },
-    _checkAndInstall
+    async (ctx: TelemetryContext) => {
+      const res = await _checkAndInstall("LocalDebug Prerequisite Check", orderedCheckers, ports);
+      if (res.error) {
+        const debugSession = commonUtils.getLocalDebugSession();
+        addCheckResultsForTelemetry(
+          res.checkResults,
+          debugSession.properties,
+          debugSession.errorProps
+        );
+        addCheckResultsForTelemetry(res.checkResults, ctx.properties, ctx.errorProps);
+        return err(res.error);
+      }
+      return ok(undefined);
+    }
   );
 }
 
-async function _checkAndInstall(ctx: TelemetryContext): Promise<Result<void, FxError>> {
+// TODO: use _checkAndInstall for new prerequisite task
+// export async function checkAndInstallV2(
+//   orderedCheckers: PrerequisiteOrderedChecker[],
+//   ports: number[]
+// ): Promise<Result<void, FxError>> {
+// }
+
+async function _checkAndInstall(
+  logLabel: string,
+  orderedCheckers: PrerequisiteOrderedChecker[],
+  ports: number[]
```

Comment:
It's weird to have `ports` as input. `ports` is bound with port checker, then what if `orderedCheckers` does not contain port checker?
Is it possible to move `ports` into `checkPort`?

Reply:
I try to move all the configuration related to `projectSettings` out of checker. What about move the args (ports) into PrerequisiteOrderedChecker?